src/clarity-angular/data/datagrid/datagrid-row.ts

```diff
@@ -65,7 +74,11 @@ let nbRow: number = 0;
             <ng-content select="clr-dg-row-detail"></ng-content>
         </ng-template>
     `,
-    host: {"[class.datagrid-row]": "true", "[class.datagrid-selected]": "selected"},
+    host: {
+        "[class.datagrid-row]": "true",
+        "[class.datagrid-selected]": "selected",
+        "[attr.tabindex]": "selection.rowSelectionMode ? 0 : null"
```

Comment:
You're still missing `role="button"`. 😉 
You made it tabbable but didn't indicate it was clickable.

To be honest, the best solution would be to have some ARIA attribute that lets you indicate the full row as a `label` for the checkbox or radio inside of it, but I could find one after a quick research. Maybe you might have heard of one.

Reply:
Added the `role="button"` attribute and implemented the selection with space and enter.
This is the recommended approach.